salt/state.py

```diff
@@ -2813,29 +2855,15 @@ def render_state(self, sls, saltenv, mods, matches, local=False):
                         continue
 
                     if inc_sls.startswith('.'):
-                        match = re.match(r'^(\.+)(.*)$', inc_sls)
-                        if match:
-                            levels, include = match.groups()
-                        else:
-                            msg = ('Badly formatted include {0} found in include '
-                                    'in SLS \'{2}:{3}\''
-                                    .format(inc_sls, saltenv, sls))
-                            log.error(msg)
-                            errors.append(msg)
-                            continue
-                        level_count = len(levels)
-                        p_comps = sls.split('.')
-                        if state_data.get('source', '').endswith('/init.sls'):
-                            p_comps.append('init')
-                        if level_count > len(p_comps):
-                            msg = ('Attempted relative include of \'{0}\' '
-                                   'within SLS \'{1}:{2}\' '
-                                   'goes beyond top level package '
-                                   .format(inc_sls, saltenv, sls))
-                            log.error(msg)
-                            errors.append(msg)
+                        try:
+                            source_file = state_data.get('source', '')
+                            _sls = sls
+                            if source_file.endswith('/init.sls'):
+                                _sls += '.init'
```

Comment:
`fn_` is what I need in this function, but on [line 1303](https://github.com/saltstack/salt/pull/32469/files#diff-e5cf22153489a6c40864babf3ca18002R1303), we don't seem have that information, since we're dealing with the high data directly.

`body['__sls__']` will contain `foo` for both `foo.sls` and `foo/init.sls`, but a relative import of `.bar` in the first case would be `bar`, in the second case would be `foo.bar`.

It would help to track more metadata about which files each high block came from, but it might not be worth it if it's only for this feature. Is there something I'm missing, or any preferences regarding which direction to take this?